sql/logictest: TestParallel failed

sql/logictest.TestParallel [failed](https://tanzanite.cluster.engflow.com/invocations/default/265c0d3c-2c49-4572-a9a0-cd4b4bbf6278?testReportRun=2&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvc3FsL2xvZ2ljdGVzdDpsb2dpY3Rlc3RfdGVzdA==) on master @ [c994982a8be5af89f594e115e897dd6d62cf99d8](https://github.com/cockroachdb/cockroach/commits/c994982a8be5af89f594e115e897dd6d62cf99d8):


```
github.com/cockroachdb/cockroach/pkg/kv/kvserver/concurrency/lock_table.go:4631 concurrency.(*lockTableImpl).verify ???
github.com/cockroachdb/cockroach/pkg/kv/kvserver/concurrency/verifiable_lock_table.go:98 concurrency.verifyingLockTable.UpdateLocks ???
github.com/cockroachdb/cockroach/pkg/kv/kvserver/concurrency/concurrency_manager.go:554 concurrency.(*managerImpl).OnLockUpdated ???
github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_proposal.go:828 kvserver.(*Replica).handleReadWriteLocalEvalResult ???
github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_application_state_machine.go:233 kvserver.(*replicaStateMachine).ApplySideEffects ???
github.com/cockroachdb/cockroach/pkg/kv/kvserver/apply/cmd.go:214 apply.mapCheckedCmdIter ???
github.com/cockroachdb/cockroach/pkg/kv/kvserver/apply/task.go:295 apply.(*Task).applyOneBatch ???
github.com/cockroachdb/cockroach/pkg/kv/kvserver/apply/task.go:250 apply.(*Task).ApplyCommittedEntries ???
github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_application_state_machine.go:453 kvserver.(*replicaStateMachine).moveStats ???
github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_raft.go:749 kvserver.(*Replica).handleRaftReady ???
github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/store_raft.go:688 kvserver.(*Store).processReady ???
github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/scheduler.go:420 kvserver.(*raftSchedulerShard).worker ???
github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/scheduler.go:321 kvserver.(*raftScheduler).Start.func2 ???
github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:484 stop.(*Stopper).RunAsyncTaskEx.func2 ???

goroutine 11191283 lock 0xc0139f8cd0
github.com/cockroachdb/cockroach/pkg/util/tracing/crdbspan.go:1447 tracing.(*crdbSpan).SetRecordingType ??? <<<<<
github.com/cockroachdb/cockroach/pkg/util/tracing/crdbspan.go:1446 tracing.(*crdbSpan).SetRecordingType ???
github.com/cockroachdb/cockroach/pkg/util/tracing/span.go:841 tracing.(*spanRef).empty ???
github.com/cockroachdb/cockroach/pkg/util/tracing/tracer.go:1092 tracing.(*Tracer).StartSpanCtx ???
github.com/cockroachdb/cockroach/pkg/util/tracing/tracer.go:1629 tracing.EnsureChildSpan ???
github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:349 sql.(*connExecutor).execStmtInOpenState ???
github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:145 sql.(*connExecutor).execStmt.func1 ???
github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:3491 sql.(*connExecutor).execWithProfiling ???
github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:144 sql.(*connExecutor).execStmt ???
github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:2287 sql.(*connExecutor).execCmd.func1 ???
github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:2292 sql.(*connExecutor).execCmd ???
github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:2209 sql.(*connExecutor).run ???
github.com/cockroachdb/cockroach/pkg/sql/internal.go:232 sql.(*InternalExecutor).runWithEx.func1 ???

goroutine 33758 lock 0xc0099ef898
github.com/cockroachdb/cockroach/pkg/kv/kvserver/concurrency/lock_table.go:4638 concurrency.(*lockTableImpl).verify.func1 ??? <<<<<
github.com/cockroachdb/cockroach/pkg/kv/kvserver/concurrency/lock_table.go:4637 concurrency.(*lockTableImpl).verify.func1 ???
github.com/cockroachdb/cockroach/pkg/kv/kvserver/concurrency/lock_table.go:4641 concurrency.(*lockTableImpl).verify ???
github.com/cockroachdb/cockroach/pkg/kv/kvserver/concurrency/verifiable_lock_table.go:98 concurrency.verifyingLockTable.UpdateLocks ???
github.com/cockroachdb/cockroach/pkg/kv/kvserver/concurrency/concurrency_manager.go:554 concurrency.(*managerImpl).OnLockUpdated ???
github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_proposal.go:828 kvserver.(*Replica).handleReadWriteLocalEvalResult ???
github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_application_state_machine.go:233 kvserver.(*replicaStateMachine).ApplySideEffects ???
github.com/cockroachdb/cockroach/pkg/kv/kvserver/apply/cmd.go:214 apply.mapCheckedCmdIter ???
github.com/cockroachdb/cockroach/pkg/kv/kvserver/apply/task.go:295 apply.(*Task).applyOneBatch ???
github.com/cockroachdb/cockroach/pkg/kv/kvserver/apply/task.go:250 apply.(*Task).ApplyCommittedEntries ???
github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_application_state_machine.go:453 kvserver.(*replicaStateMachine).moveStats ???
github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_raft.go:749 kvserver.(*Replica).handleRaftReady ???
github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/store_raft.go:688 kvserver.(*Store).processReady ???
github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/scheduler.go:420 kvserver.(*raftSchedulerShard).worker ???
github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/scheduler.go:321 kvserver.(*raftScheduler).Start.func2 ???
github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:484 stop.(*Stopper).RunAsyncTaskEx.func2 ???



```

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=2</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-queries
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestParallel.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>


Jira issue: CRDB-36980

sql/logictest.TestParallel [failed](https://tanzanite.cluster.engflow.com/invocations/default/d863d545-fae5-4177-a0f7-5ec8a17834d4?testReportRun=3&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvc3FsL2xvZ2ljdGVzdDpsb2dpY3Rlc3RfdGVzdA==) on master @ [7fc4c7bcbbf0c75a62d056da0bf79a5a32714650](https://github.com/cockroachdb/cockroach/commits/7fc4c7bcbbf0c75a62d056da0bf79a5a32714650):


```
	NOTICE: .ClusterSettings() called via implicit interface ApplicationLayerInterface;
HINT: consider using .ApplicationLayer().ClusterSettings() instead.
TIP: consider replacing the test server initialization from:
    ts, ... := serverutils.StartServer(t, ...)
    defer ts.Stopper().Stop(...)
to:
    srv, ... := serverutils.StartServer(t, ...)
    defer srv.Stopper().Stop(...)
    ts := srv.ApplicationLayer()

See also: https://go.crdb.dev/p/testserver-and-cluster-virtualization

pkg/sql/logictest/parallel_test.go:178: ((*parallelTest).setup)
	NOTICE: .ClusterSettings() called via implicit interface ApplicationLayerInterface;
HINT: consider using .ApplicationLayer().ClusterSettings() instead.
TIP: consider replacing the test server initialization from:
    ts, ... := serverutils.StartServer(t, ...)
    defer ts.Stopper().Stop(...)
to:
    srv, ... := serverutils.StartServer(t, ...)
    defer srv.Stopper().Stop(...)
    ts := srv.ApplicationLayer()

See also: https://go.crdb.dev/p/testserver-and-cluster-virtualization
[07:56:40] --- done: testdata/parallel_test/subquery_retry_multinode/setup with config : 2 tests, 0 failures
[07:56:40] --- done: testdata/parallel_test/subquery_retry_multinode/txn with config : 1 tests, 0 failures
[07:56:40] --- done: testdata/parallel_test/subquery_retry_multinode/txn with config : 0 tests, 0 failures
    parallel_test.go:85: 
        testdata/parallel_test/subquery_retry_multinode/txn:9: 
        expected success, but found
        (40003) result is ambiguous: error=TransactionRetryError: retry txn (RETRY_SERIALIZABLE): "sql txn" meta={id=69eba613 key=/Tenant/10/Table/106/1/956457944935727106/0 iso=Serializable pri=0.01993827 epo=0 ts=1711958200.575203941,1 min=1711958200.476034268,0 seq=3} lock=true stat=PENDING rts=1711958200.571955961,0 wto=false gul=1711958200.976034268,0 [propagate] (last error: TransactionRetryError: retry txn (RETRY_SERIALIZABLE): "sql txn" meta={id=69eba613 key=/Tenant/10/Table/106/1/956457944935727106/0 iso=Serializable pri=0.01993827 epo=0 ts=1711958200.590019009,1 min=1711958200.476034268,0 seq=3} lock=true stat=PENDING rts=1711958200.571955961,0 wto=false gul=1711958200.976034268,0)
        dist_sender.go:3057: in sendToReplicas()
[07:56:40] --- done: testdata/parallel_test/subquery_retry_multinode/txn with config : 0 tests, 0 failures
    parallel_test.go:85: 
        testdata/parallel_test/subquery_retry_multinode/txn:9: 
        expected success, but found
        (40003) result is ambiguous: error=TransactionRetryError: retry txn (RETRY_SERIALIZABLE): "sql txn" meta={id=da0d3ab6 key=/Tenant/10/Table/106/1/956457944972361732/0 iso=Serializable pri=0.03565797 epo=0 ts=1711958200.590019009,1 min=1711958200.475494099,0 seq=3} lock=true stat=PENDING rts=1711958200.583434898,0 wto=false gul=1711958200.975494099,0 [propagate] (last error: TransactionRetryError: retry txn (RETRY_SERIALIZABLE): "sql txn" meta={id=da0d3ab6 key=/Tenant/10/Table/106/1/956457944972361732/0 iso=Serializable pri=0.03565797 epo=0 ts=1711958200.590019009,1 min=1711958200.475494099,0 seq=3} lock=true stat=PENDING rts=1711958200.583434898,0 wto=false gul=1711958200.975494099,0)
        dist_sender.go:3057: in sendToReplicas()
[07:56:40] --- done: testdata/parallel_test/subquery_retry_multinode/txn with config : 0 tests, 0 failures
    parallel_test.go:85: 
        testdata/parallel_test/subquery_retry_multinode/txn:9: 
        expected success, but found
        (40003) result is ambiguous: error=TransactionRetryError: retry txn (RETRY_SERIALIZABLE): "sql txn" meta={id=3fbd5284 key=/Tenant/10/Table/106/1/956457944951357443/0 iso=Serializable pri=0.04357616 epo=0 ts=1711958200.583434898,1 min=1711958200.475891476,0 seq=3} lock=true stat=PENDING rts=1711958200.575203941,0 wto=false gul=1711958200.975891476,0 [propagate] (last error: TransactionRetryError: retry txn (RETRY_SERIALIZABLE): "sql txn" meta={id=3fbd5284 key=/Tenant/10/Table/106/1/956457944951357443/0 iso=Serializable pri=0.04357616 epo=0 ts=1711958200.590019009,1 min=1711958200.475891476,0 seq=3} lock=true stat=PENDING rts=1711958200.575203941,0 wto=false gul=1711958200.975891476,0)
        dist_sender.go:3057: in sendToReplicas()
[07:56:40] --- done: testdata/parallel_test/subquery_retry_multinode/txn with config : 1 tests, 0 failures
[07:56:40] --- done: testdata/parallel_test/subquery_retry_multinode/final with config : 2 tests, 0 failures
    parallel_test.go:253: -- test log scope end --
test logs left over in: outputs.zip/logTestParallel_subquery_retry_multinode1866448122
I240401 07:56:40.928423 13558 sql/logictest/parallel_test.go:258  [-] 2  1 out of 3 parallel tests failed
    --- FAIL: TestParallel/subquery_retry_multinode (27.25s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=3</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #121168 sql/logictest: TestParallel failed [24.1 A-kv-client C-bug C-test-failure O-robot P-1 T-kv release-blocker]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestParallel.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

--- a/pkg/sql/logictest/parallel_test.go
+++ b/pkg/sql/logictest/parallel_test.go
@@ -230,6 +230,7 @@ func TestParallel(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 
 	skip.UnderRace(t, "takes >1 min under race")
+	skip.UnderDeadlock(t, "too slow")
 
 	glob := *paralleltestdata
 	paths, err := filepath.Glob(glob)

